iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -396,45 +396,49 @@ public static PartitionData toPartitionData(StructLike sourceKey, Types.StructTy
 
   /**
    * Returns list of data files filtered by specId and partitionPath as following:
-   *  1. If matchBySpecId is true, then filters files by specId == file's specId, else by specId != file's specId
-   *  2. If partitionPath is not null, then also filters files where partitionPath == file's partition path
+   *  1. If table is unpartitioned, returns all data files without filtering.
+   *  2. If matchBySpecId is true, then filters files by specId == file's specId, else by specId != file's specId
+   *  3. If partitionPath is not null, then also filters files where partitionPath == file's partition path
    * @param table the iceberg table
    * @param specId partition spec id
    * @param partitionPath partition path
    * @param matchBySpecId filter that's applied on data files' spec ids
    */
-  public static List<DataFile> getDataFiles(Table table, int specId, String partitionPath,
+  public static List<DataFile> getDataFiles(Table table, Integer specId, String partitionPath,
       Predicate<Object> matchBySpecId) {
     CloseableIterable<FileScanTask> fileScanTasks =
         table.newScan().useSnapshot(table.currentSnapshot().snapshotId()).ignoreResiduals().planFiles();
     CloseableIterable<FileScanTask> filteredFileScanTasks =
         CloseableIterable.filter(fileScanTasks, t -> {
           DataFile file = t.asFileScanTask().file();
-          return matchBySpecId.test(file.specId()) && (partitionPath == null || (partitionPath != null &&
-                  table.specs().get(specId).partitionToPath(file.partition()).equals(partitionPath)));
+          return !table.spec().isPartitioned() ||
+              (matchBySpecId.test(file.specId()) && (partitionPath == null || (partitionPath != null &&
```

Comment:
can't we remove `partitionPath != null` check from OR 2nd operand?

Reply:
Done

Reply:
@difin please check the comment, I've updated the suggestion

Reply:
Done